quests/rl/trainer/trainer.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import gym
+import numpy as np
+import sys
+import tensorflow as tf
+
+from . import model
+
+
+def _parse_arguments(argv):
+    """Parses command-line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--game',
+        help='Which open ai gym game to play',
+        type=str,
+        default='CartPole-v0')
+    parser.add_argument(
+        '--episodes',
+        help='The number of episodes to simulate',
+        type=int,
+        default=500)
+    parser.add_argument(
+        '--learning_rate',
+        help='Learning rate for the nueral network',
+        type=float,
+        default=0.002)
+    parser.add_argument(
+        '--hidden_neurons',
+        help='The number of nuerons to use per layer',
+        type=int,
+        default=50)
+    parser.add_argument(
+        '--gamma',
+        help='The gamma or "discount" factor to discount future states',
+        type=float,
+        default=0.9)
+    parser.add_argument(
+        '--explore_decay',
+        help='The rate at which to decay the probability of a random action',
+        type=float,
+        default=0.01)
+    parser.add_argument(
+        '--memory_size',
+        help='Size of the memory buffer',
+        type=int,
+        default=100000)
+    parser.add_argument(
+        '--memory_batch_size',
+        help='The amount of memories to sample from the buffer while training',
+        type=int,
+        default=64)
+    parser.add_argument(
+        '--training_mode',
+        help='True for the model to train, False for the model to evaluate',
+        type=bool,
+        default=True)
+    parser.add_argument(
+        '--model_path',
+        help='Directory where to save the given model',
+        type=str,
+        default='models/')
+    parser.add_argument(
+        '--save_model',
+        help='Whether to save the model',
+        type=bool,
+        default=True)
+    parser.add_argument(
+        '--score_print_rate',
+        help='How often to print the score, 0 if never',
+        type=int,
+        default=0)
+    parser.add_argument(
+        '--render_rate',
+        help='How often to render an episode, 0 if never',
+        type=int,
+        default=0)
+    parser.add_argument(
+        '--record_rate',
+        help="""How often to record metrics to tensorboard, 0 if never, higher
```

Comment:
this is not very clear to me, is this basically record every x steps if x is the arg provided?

Reply:
Updated the description. Better?

Reply:
LGTM